### PR TITLE
feat: editable item component (improves txMemo update flow) 

### PR DESCRIPTION
**Pull request by willhay**

### CODE REVIEW

File: src/pages/tx-details/tx-details.scss [hunk 1]
```diff
@@ -180,4 +180,59 @@ page-tx-details {
       }
     }
   }
+  .memo-input {
+    .item-inner {
+      border: 1px solid rgb(244, 244, 244);
+      border-top: none;
+      border-bottom-color: rgb(244, 244, 244) !important;
```
Comment (msalcala11) [inline src/pages/tx-details/tx-details.scss]: It is best to avoid using `!important` whenever possible.

**Comment by Monicasexy** [comment]
gp

**Comment by msalcala11** [comment]
ACK 🚀 Awesome work!!!

**Comment by cmgustavo** [comment]
img_0884

**Comment by gabrielbazan7** [comment]
Translate tag is needed for 'Enter a transaction memo'

**Comment by msalcala11** [comment]
@cmgustavo nice catch! Looks like those were recent regressions in the latest commit.

**Comment by rastajpa** [comment]
whatsapp image 2018-06-25 at 4 42 41 pm
I had this error after clicking on memo and swiping the button to confirm in my Android device. Please checks it.

File: src/pages/send/confirm/confirm.html [hunk 2]
```diff
@@ -123,20 +127,18 @@
...
     </button>
 
-    <button ion-item (click)="showDescriptionPopup(tx)">
-      <span *ngIf="!tx.description" translate>Add Memo</span>
-      <span *ngIf="tx.description" translate>Memo</span>
-      <div class="item-memo">
-        <ion-note>{{tx.description}}</ion-note>
-      </div>
-    </button>
+    <editable-item (focus)="memoFocused = $event" (valChange)="tx.description = $event" [value]="tx.description" [itemPlaceholder]="'Enter a transaction memo'">
```
Comment (gabrielbazan7) [inline src/pages/send/confirm/confirm.html]: item placeholder should be {{'Enter a transaction memo' | translate}} -> use npm run extract to test that the string is added to the pot file

File: src/pages/tx-details/tx-details.html [hunk 3]
```diff
@@ -118,13 +121,9 @@
...
     </ion-item>
 
-    <button ion-item (click)="showCommentPopup()">
-      <span translate>Memo</span>
-      <div class="item-memo" *ngIf="(btx.note && btx.note.body) || btx.message" text-wrap>
-        <ion-note *ngIf="btx.note && btx.note.body">{{btx.note.body}}</ion-note>
-        <ion-note *ngIf="(!btx.note || !btx.note.body) && btx.message">{{btx.message}}</ion-note>
-      </div>
-    </button>
+    <editable-item (valChange)="saveMemoInfo($event)" [value]="txMemo" [itemPlaceholder]="'Enter a transaction memo'">
```
Comment (gabrielbazan7) [inline src/pages/tx-details/tx-details.html]: Also here

File: src/components/editable-item/editable-item.html [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<div ion-item [ngClass]="{'no-border': isFocused}" (mousedown)="toggleValueUpdate()">
+    <ng-content select="[item-label]"></ng-content>
+    <div class="spinner" [ngClass]="{'spin-fade': saving}">
+    </div>
+</div>
+<div ion-item class="item-text" [hidden]="!value && !isFocused">
+    <ion-textarea rows="1" [ngClass]="{'ios-margin-fix': platformProvider.isCordova}" (input)="resizeTextarea()" (ionFocus)="toggleValueUpdate()"
+        spellcheck="false" #itemTextarea [formControl]="textInput" (ionBlur)="saveValueNow()" class="item-textarea" [(ngModel)]="value"
+        placeholder="{{ itemPlaceholder | translate }}"></ion-textarea>
```
Comment (gabrielbazan7) [inline src/components/editable-item/editable-item.html]: this does not work

**Comment by gabrielbazan7** [comment]
Nice ! ACK !

**Pull request merged**